lib/hs/src/Thrift/Server.hs

```diff
@@ -42,25 +42,41 @@ runThreadedServer :: (Protocol i, Protocol o)
                   => (Socket -> IO (i, o))
                   -> h
                   -> (h -> (i, o) -> IO Bool)
-                  -> PortID
+                  -> Int
                   -> IO a
 runThreadedServer accepter hand proc_ port = do
-    socket <- listenOn port
-    acceptLoop (accepter socket) (proc_ hand)
+    let hints = defaultHints {
```

Comment:
This is basically copy-paste from the network library how-to on how to listen on a socket.